packages/generator/ni_measurement_plugin_sdk_generator/client/__init__.py

```diff
@@ -161,30 +168,34 @@ def _create_clients_interactively() -> None:
             int(selection), measurement_service_classes
         )
 
-        base_service_class = extract_base_service_class(service_class)
-        default_module_name = create_module_name(base_service_class, generated_modules)
-        module_name = click.prompt(
-            "Enter a name for the Python client module, or press Enter to use the default name.",
-            type=str,
-            default=default_module_name,
-        )
-        validate_identifier(module_name, "module")
-        default_class_name = create_class_name(base_service_class)
-        class_name = click.prompt(
-            "Enter a name for the Python client class, or press Enter to use the default name.",
-            type=str,
-            default=default_class_name,
-        )
-        validate_identifier(class_name, "class")
-
-        _create_client(
-            channel_pool=channel_pool,
-            discovery_client=discovery_client,
-            measurement_service_class=service_class,
-            module_name=module_name,
-            class_name=class_name,
-            directory_out=directory_out_path,
-        )
+        try:
+            base_service_class = extract_base_service_class(service_class)
+            default_module_name = create_module_name(base_service_class, generated_modules)
+            module_name = click.prompt(
+                "Enter a name for the Python client module, or press Enter to use the default name.",
+                type=str,
+                default=default_module_name,
+            )
+            validate_identifier(module_name, "module")
+            default_class_name = create_class_name(base_service_class)
+            class_name = click.prompt(
+                "Enter a name for the Python client class, or press Enter to use the default name.",
+                type=str,
+                default=default_class_name,
+            )
+            validate_identifier(class_name, "class")
+
+            _create_client(
+                channel_pool=channel_pool,
+                discovery_client=discovery_client,
+                measurement_service_class=service_class,
+                module_name=module_name,
+                class_name=class_name,
+                directory_out=directory_out_path,
+            )
+        except Exception as e:
+            click.echo(CLIENT_CREATION_ERROR_MESSAGE.format(service_class, e))
+            break
         generated_modules.append(module_name)
```

Comment:
The current logic appends the module name even when client generation fails. Therefore, this should be moved inside the try block.